### DISCUSSION TITLE
Enable Image Optimization in @astrojs/vercel

### DISCUSSION THREAD

**Discussion by joe-bell** [withastro/roadmap#448] [Proposal]
# Summary

Allow consumers of `@astrojs/vercel` to take advantage of its [**built-in Image Optimization** offering](https://vercel.com/docs/concepts/image-optimization#usage)

# Background & Motivation

Currently, Astro recommends developers to use `@astrojs/image`, but this comes with a key limitation: 

It isn't supported in the "edge" build target.

For Vercel customers, this problem could be avoided altogether by using Vercel's image optimization offering

# Goals

- Expose Vercel's [Build Output API](https://vercel.com/docs/build-output-api/v3#build-output-configuration/supported-properties/images/example) options to `@astrojs/vercel` for Vercel customers to leverage image optimization without needing a separate package
- Unblock developers who want to make use of image optimization in an edge environment

## Caveats

- Developers will need to run `dev` via `vercel dev` to see the images locally (unless of course `astro dev` could take care of this for us 🤔)

# Example

## Configuration

```ts
// astro.config.ts
import { defineConfig } from "astro/config";
import vercel from "@astrojs/vercel/edge";

export default defineConfig({
  output: "sever",
  adapter: vercel({
    images: {
      sizes: [640, 750, 828, 1080, 1200],
      domains: [],
      minimumCacheTTL: 60,
      formats: ["image/avif", "image/webp"],
      remotePatterns: [
        {
          protocol: "https",
          hostname: "^via\\.placeholder\\.com$",
          pathname: "^/1280x640/.*$",
        },
      ],
    },
  }),
});
```

## Usage

```astro
// image.astro
---
const { src } = Astro.props;
---

<img src={`/_vercel/image?url=${src}`} />
```

**Comment by andrewdoro** [comment]
This guy actually wrote an article about making this
https://jcbl.ws/astro-vercel-image-optimization

**Comment by Princesseuh** [comment]
I am currently working on implementing this into `astro:assets`. Here's the PR to follow: https://github.com/withastro/astro/pull/6845

Since this is specific to one integration, we won't need to go through the full RFC process.

**Comment by Princesseuh** [comment]
Support for this has been merged and will be available in the next minor of the Vercel adapter, thank you everyone!

**Reply by joe-bell** [reply]
Thank you!
